### PR TITLE
[compiler-v2] Stop stackless bytecode pipeline when errors are found by a processor

### LINKED ISSUE #14506: [Bug][compiler-v2] V2 crashes when using unassigned reference
# 🐛 Bug

```
//# publish
module 0xCAFE::Module0 {
    struct S has copy, drop {}
    public fun function0() {
        let y: &S;
        *y;
    }
}
```

This will cause compiler V2 to panic:
```
thread 'main' panicked at .../aptos-core/third_party/move/move-borrow-graph/src/graph.rs:61:44:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

V1 will give the expected error message `error[E06002]: use of unassigned variable`.

### PR DESCRIPTION
**Pull request by vineethk**
## Description

Fixes #14506.

Stackless bytecode pipeline is made of a series of checkers and optimizers, and subsequent checkers and optimizer passes could assume that certain errors are not present if you have reached this processor. In this PR, we stop the stackless bytecode pipeline when errors are found by a processor.

Until we have tested each pass explicitly for various kinds of errors they can sustain, we should stop after a processor finds errors.

Both the compiler-v2 and its testing now have the same behavior: stackless bytecode pipeline is stopped early if errors are found by a processor in the pipeline.

## Type of Change
- [x] Bug fix

## Which Components or Systems Does This Change Impact?
- [x] Move/Aptos Virtual Machine

## How Has This Been Tested?
* Existing tests pass, baseline updates have been reviewed, some tests were split into two to ensure previously reported errors are continued to be reported
* New tests have been added to show the compiler crash being fixed.

## Key Areas to Review
The logic to stop early on finding errors.

### CODE REVIEW

**Comment by vineethk** [comment]
* **#14524** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/14524?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vineethk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: aptos-move/aptos-transactional-test-harness/tests/aptos_test_harness/diamond_clicker.v2_exp [hunk 1]
```diff
@@ -33,26 +33,26 @@ B0:
 	9: MutBorrowField[0](OuterStruct.any_field: vector<InnerStruct>)
 	10: StLoc[2](loc1: &mut vector<InnerStruct>)
 	11: LdU64(0)
-	12: StLoc[3](loc2: u64)
```
Comment (vineethk) [inline aptos-move/aptos-transactional-test-harness/tests/aptos_test_harness/diamond_clicker.v2_exp:36]: These changes are unrelated to this PR, but were obtained by running `move_pr.sh -i2`.

File: aptos-move/aptos-transactional-test-harness/tests/aptos_test_harness/table.v2_exp [hunk 2]
```diff
@@ -52,7 +52,7 @@ Events:
 }
 {
     type:    0x1::transaction_fee::FeeStatement
-    data:    "9c0000000000000006000000000000000400000000000000c83a0200000000000000000000000000"
```
Comment (vineethk) [inline aptos-move/aptos-transactional-test-harness/tests/aptos_test_harness/table.v2_exp:55]: These changes are unrelated to this PR, but were obtained by running `move_pr.sh -i2`.

File: third_party/move/move-model/tests/sources/structs_err.v2_exp [hunk 3]
```diff
@@ -22,11 +22,11 @@ error: missing fields `y`, `z`
 40 │       S{x: x}
    │       ^^^^^^^
 
-error: cannot return `G<u64>` from a function with result type `G<bool>`
```
Comment (vineethk) [inline third_party/move/move-model/tests/sources/structs_err.v2_exp:25]: These changes are unrelated to this PR, but were obtained by running `move_pr.sh -i2`.

**Review by rahxephon89 — APPROVED** [review]

File: third_party/move/move-compiler-v2/tests/testsuite.rs [hunk 4]
```diff
@@ -810,6 +810,7 @@ fn run_test(path: &Path, config: TestConfig) -> datatest_stable::Result<()> {
                         out.push_str(dump);
```
[the remark this answers is not in the document]
Comment (vineethk) [reply third_party/move/move-compiler-v2/tests/testsuite.rs:810]: Done.

**Pull request merged**